cpp/velox/substrait/SubstraitToVeloxPlanValidator.cc

```diff
@@ -150,9 +151,18 @@ bool SubstraitToVeloxPlanValidator::validateScalarFunction(
   if (name == "extract") {
     return validateExtractExpr(params);
   }
-  if (name == "regexp_extract_all" && !scalarFunction.arguments()[1].value().has_literal()) {
-    logValidateMsg("native validation failed due to: pattern is not literal for regex_extract_all.");
-    return false;
+  if (name == "regexp_extract_all" || name == "regexp_extract" || name == "regexp_replace" || name == "rlike") {
+    auto patternArg = scalarFunction.arguments()[1].value();
+    if (!patternArg.has_literal()) {
+      logValidateMsg("native validation failed due to: pattern is not literal for regex_extract_all.");
+      return false;
+    }
+    auto pattern = patternArg.literal().string();
+    auto re2 = RE2(pattern, RE2::Quiet);
+    if (!re2.ok()) {
+      logValidateMsg("native validation failed due to: pattern " + pattern + " compilation failed in RE2.");
```

Comment:
Can we expose RE2's compilation error message to the validation message? It may be helpful to users to figure out the root reason. Thanks!

Reply:
Weird but the RE2 error message is always empty.

Reply:
Maybe, `Quiet` option makes it silent. I'm not sure. You can try to change it. Thanks!

Reply:
Tried default options with log print but not help. I'm not sure if it's an issue from the current RE2 version we use.

Reply:
Just found it's due to my local env was broken. Already added.